client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts

```diff
@@ -129,6 +132,19 @@ export function getEnhancedTasks(
 						},
 					};
 					break;
+				case 'unlock_styles':
+					if ( ! isEnabled( 'limit-global-styles' ) || ! globalStylesInUse ) {
+						return;
+					}
+					taskData = {
+						title: translate( 'Activate advanced design customization tools' ),
+						isCompleted: ! shouldLimitGlobalStyles,
+						actionDispatch: () => {
+							recordTaskClickTracksEvent( flow, task.isCompleted, task.id );
+							window.location.replace( `/site-editor/${ siteSlug }?unlock-styles` );
```

Comment:
If the current URL has query parameters that need to propagate, we'll be removing them from the replaced URL.

Reply:
AFAIK we don't need to propagate any query param, since we're changing the context and the site editor wouldn't know what to do with them. This is a usual approach we apply everywhere else. See some example below:

https://github.com/Automattic/wp-calypso/blob/d0f2d83b8c6e42973c801fbfe219d22ecadaf503/client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts#L52

https://github.com/Automattic/wp-calypso/blob/d0f2d83b8c6e42973c801fbfe219d22ecadaf503/client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts#L74

https://github.com/Automattic/wp-calypso/blob/d0f2d83b8c6e42973c801fbfe219d22ecadaf503/client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts#L100

https://github.com/Automattic/wp-calypso/blob/d0f2d83b8c6e42973c801fbfe219d22ecadaf503/client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts#L124